packages/scm/src/browser/scm-contribution.ts

```diff
@@ -0,0 +1,99 @@
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { inject, injectable } from 'inversify';
+import {
+    AbstractViewContribution,
+    FrontendApplication,
+    FrontendApplicationContribution,
+    QuickOpenService,
+    StatusBar,
+    StatusBarAlignment,
+    StatusBarEntry
+} from '@theia/core/lib/browser';
+import {ScmCommand, ScmService} from './scm-service';
+import { ScmWidget } from '../browser/scm-widget';
+import URI from '@theia/core/lib/common/uri';
+import {CommandRegistry} from '@theia/core';
+import {ScmQuickOpenService} from './scm-quick-open-service';
+
+export const SCM_WIDGET_FACTORY_ID = 'scm';
+
+@injectable()
+export class ScmContribution extends AbstractViewContribution<ScmWidget> implements FrontendApplicationContribution {
+
+    @inject(StatusBar) protected readonly statusBar: StatusBar;
+    @inject(ScmService) protected readonly scmService: ScmService;
+    @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry;
+    @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService;
+    @inject(ScmQuickOpenService) protected readonly scmQuickOpenService: ScmQuickOpenService;
+    constructor() {
+        super({
+            widgetId: SCM_WIDGET_FACTORY_ID,
+            widgetName: 'Scm',
+            defaultWidgetOptions: {
+                area: 'left',
+                rank: 300
+            },
+            toggleCommandId: 'scmView:toggle'
+        });
+    }
+
+    onStart(): void {
+        const CHANGE_REPOSITORY = {
+            id: 'scm.change.repository',
+            label: 'Scm: Change Repository...'
+        };
+
+        const refresh = (commands: ScmCommand[]) => {
+            commands.forEach(command => {
+                const statusBaCommand: StatusBarEntry = {
+                    text: command.text,
+                    tooltip: command.tooltip,
+                    command: command.command,
+                    alignment: StatusBarAlignment.LEFT,
+                    priority: 100
+                };
+                this.statusBar.setElement(command.id, statusBaCommand);
+            });
+        };
+        this.scmService.onDidAddRepository(repository => {
+            const onDidChangeStatusBarCommands = repository.provider.onDidChangeStatusBarCommands;
+            if (onDidChangeStatusBarCommands) {
+                onDidChangeStatusBarCommands(commands => refresh(commands));
+            }
+        });
+
+        this.commandRegistry.registerCommand(CHANGE_REPOSITORY, {
+            execute: () => {
+                this.scmQuickOpenService.changeRepository();
+            }
+        });
+        this.scmService.onDidChangeSelectedRepositories(repository => {
+            const path = new URI(repository.provider.rootUri).path;
```

Comment:
we should use `LabelProvider.getLongPath` to get human readable path

Reply:
done